src/classes/RD2_OpportunityEvaluationService.cls

```diff
@@ -251,18 +251,17 @@ public inherited sharing class RD2_OpportunityEvaluationService {
     };
 
     /**
-    * @description Contains Map of Recurring Donation Ids to Boolean indicating a schedule change
-    * @return Map<Id, Boolean> isScheduleChangedById maps Recurring Donation Id to a boolean
-    * indicating whether the Recurring Donation update results in a new schedule.
+    * @description Recurring Donation Ids for which schedule has been changed
     */
     @TestVisible
-    private Map<Id, Boolean> isScheduleChangedById {
+    private Set<Id> scheduleChangedRDIds {
```

Comment:
@force2b , @npsp-reedestockton 

Can you please preliminary review the code? There is a scenario when the Day of Month is changed that I will probably fix in this PR (details are already specified in the WI).

Also, when the schedule is changed, the Next Donation Date *initially* ignores the pause schedule, but the Opp eval service updates it correctly in async. The RD2 batch also calculates the Next Donation Date correctly.

However, in meantime can you go over the changes. I applied refactoring on the schedule changed code, ie used `Set` instead of the `Map`. Thanks.

CC @rponti-sforg